src/main/java/duke/module/Goal.java

```diff
@@ -0,0 +1,125 @@
+package duke.module;
+
+import duke.data.Storage;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Map;
+
+public class Goal {
+
+    /**
+     * A hash map which holds the optional goals of the day for any day.
+     */
+    private Map<Date, ArrayList<String>> goals;
+
+    /**
+     * Constructor for Goal objects.
+     * @param dateArrayListMap The hash map of goals of the day from loading
+     *                         the goals.txt text file.
+     */
+    public Goal(final Map<Date, ArrayList<String>> dateArrayListMap) {
+        this.goals = dateArrayListMap;
+    }
+
+    /**
+     * Shows all goals on a specific day.
+     * @param day The day to view all goals.
+     * @return A message containing all the goals of the day to be printed.
+     * @throws ParseException if the user input is in wrong format.
+     */
+    public String viewGoal(final String day) throws ParseException {
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        Date today = simpleDateFormat.parse(day);
+        String message = "";
+        boolean hasGoal = false;
+        for (Date d : goals.keySet()) {
+            if (d.equals(today)) {
+                if (!goals.get(d).isEmpty()) {
+                    hasGoal = true;
+                    for (String str : goals.get(d)) {
+                        message += str + "\n";
+                    }
+                }
+            }
+        }
+        if (!hasGoal) {
+            return "There is no goal of the day";
+        } else {
+            return message;
+        }
```

Comment:
This conditional path could flow better for readability. Consider setting the happy path first.